Recommendation for NATs and consequences of using CID (#256)
quicwg/ops-drafts: draft-ietf-quic-manageability.md

## QUIC and Network Address Translation (NAT)

Devices that perform Network Address Translation (NAT) should refrain from
parsing or otherwise using QUIC connection IDs but retain their existing
4-tuple-based operation instead.

As explained in section {{rebinding}},
QUIC connection IDs are opaque byte fields that are expressed consistently
across all QUIC versions {{QUIC-INVARIANTS}} and used by the endpoints to
survive address rebindings. While this may appear as an opportunities to

> minimiize NAT port usage or to support the work of the QUIC server by

Author: ```suggestion
minimize NAT port usage or to support the work of the QUIC server by
```

## QUIC and Network Address Translation (NAT)

This section explains how QUIC supports devices that perform Network Address

> Translation (NAT) better than other connection-oriented protocols and

Reviewer: ```suggestion
Translation (NAT) better than other connection-oriented protocols, and
```

also how the use of QUIC's connection ID for NAT can create serious problems

> for the endpoints even though it might appear attractive to simplify network

Reviewer: ```suggestion
for the endpoints even though it might appear to simplify network
```

management. In summary, devices that perform NAT should refrain from
parsing or otherwise using QUIC connection IDs but retain their existing
4-tuple-based operation instead.

As explained in section {{rebinding}},
QUIC connection IDs are opaque byte fields that are expressed consistently
across all QUIC versions {{QUIC-INVARIANTS}} and used by the endpoints to

> survive address rebindings. While this may appear as an opportunities to

Reviewer: ```suggestion
survive address rebindings. While this may appear as an opportunity to
```

minimize NAT port usage (see section {{nat-resource}}) or to support the
work of the QUIC server (see section {{nat-help}}) by

> hinding path changes, use of the connection ID by a NAT device to intentify

Reviewer: ```suggestion
hiding path changes, use of the connection ID by a NAT device to identify
```

flows will cause connection failure when the connection ID changes.
Wrongly assuming a fixed connection ID for each connection can disable
important protocol features that require connection IDs to be

> changeable and even impact security but facilitate amplification attacks.

Reviewer: ```suggestion
changeable and even impact security by facilitating amplification attacks.
```
